lib/encode.js

```diff
@@ -13,39 +21,61 @@ import {inspect} from './util.js';
- * @param {boolean} [options.compressionMode=1] - `1` to use compression mode
- *   version 1, `0` to use no compression.
- * @param {Map} [options.appContextMap] - A map of JSON-LD Context URLs and
- *   their encoded CBOR-LD values (must be values greater than 32767 (0x7FFF)).
- * @param {diagnosticFunction} [options.diagnose] - A function that, if
- * provided, is called with diagnostic information.
- *
+ * @param {number} options.varintValue - The varint value for the
+ *  registry entry associated with the resulting CBOR-LD payload. For legacy
+ *  support, use varintTagValue = -1.
+ * @param {Map} options.keywordsTable -A map of JSON-LD keywords
+ *  and their associated CBOR-LD integer values.
+ * @param {Map} options.stringTable - A map of string values and
+ *  their associated CBOR-LD integer values.
+ * @param {Map} options.urlSchemeTable - A map of URL schemes to
+ *  their associated CBOR-LD integer values.
+ * @param {Map} options.typedLiteralTable - A map of JSON-LD types
+ *  to maps that map values of that type to their associated CBOR-LD
+ *  integer values.
+ * @param {diagnosticFunction} options.diagnose - A function that, if
+ *  provided, is called with diagnostic information.
+ * @param {Map} options.appContextMap - For use with the legacy value of
+ *  varintTagValue.
+ * @param {number} options.compressionMode - For use with the legacy value of
+ *  varintValue.
  * @returns {Promise<Uint8Array>} - The encoded CBOR-LD bytes.
  */
 export async function encode({
-  jsonldDocument, documentLoader, appContextMap = new Map(),
-  compressionMode = 1, diagnose
+  jsonldDocument, documentLoader, varintValue = 'legacy',
+  keywordsTable = new Map(),
+  stringTable = new Map(),
+  urlSchemeTable = new Map(),
+  typedLiteralTable = new Map(),
+  diagnose,
+  appContextMap = new Map(),
+  compressionMode = 1
 } = {}) {
-  if(!(compressionMode === 0 || compressionMode === 1)) {
-    throw new TypeError(
-      '"compressionMode" must be "0" (no compression) or "1" ' +
-      'for compression mode version 1.');
+
+  // validate that an acceptable value for `varintValue` was passed
+  if(!((typeof varintValue === 'number' && varintValue > 0) ||
+    varintValue === 'legacy')) {
+    throw new CborldError(
+      'ERR_UNSUPPORTED_VARINT_VALUE',
+      'varintValue must be either a positive integer or `legacy`.');
   }
 
-  // 0xd9 == 11011001
-  // 110 = CBOR major type 6
-  // 11001 = 25, 16-bit tag size (65536 possible values)
-  // 0x05 = always the first 8-bits of a CBOR-LD tag
-  // compressionMode = last 8-bits of a CBOR-LD tag indicating compression type
-  const prefix = new Uint8Array([0xd9, 0x05, compressionMode]);
+  const isLegacy = varintValue === 'legacy';
+  const prefix = _getPrefix({isLegacy, compressionMode, varintValue});
+  const compressor = _createCompressor({
+    isLegacy,
+    appContextMap,
+    documentLoader,
+    keywordsTable,
+    stringTable,
+    typedLiteralTable,
+    urlSchemeTable
+  });
```

Comment:
This is excellent now, thanks! (And so is the simple code below to finish out the function).